mainapp/views.py

```diff
@@ -31,25 +28,47 @@ def create(request):
             prod.is_active = 1
             prod.author = Profile.objects.get(user=request.user)
             prod.save()
-        return render(request, 'mainapp/home.html'  )
+        return render(request, 'mainapp/home.html')
     else:
         form = ProductForm()
         return render(request, 'mainapp/create.html', {'form': form})
 
 
 def comment(request):
-    com = Comment.objects.all()
-    return render(request, 'mainapp/comment.html', {'title': 'Коментарccc', 'comm': com})
+    prod = request.GET.get('')
+    if request.method == 'POST':
+        form = CommentForm(request.POST)
+        if form.is_valid() and request.user.is_authenticated:
+            comm = form.save(commit=False)
+            comm.user_id = Profile.objects.get(user=request.user)
+            print(comm, comm.user_id)
+            comm.product_id = prod
+            comm.save()
+        return render(request, 'mainapp/home.html')
+    else:
+        form = CommentForm()
+        return render(request, 'mainapp/comment.html', {'form': form})
 
 
-def look(request):
-    count = User.objects.count()
-    return render(request, 'mainapp/look.html', {'count': count})
+@login_required()
+def look(request, id='1'):
+    user_list = Product.objects.get(id=id)
+    if request.method == 'POST':
+        form = CommentForm(request.POST)
+        if form.is_valid() and request.user.is_authenticated:
```

Comment:
Move `request.user.is_authenticated` to `request.method` check.

Reply:
ok